### ISSUE TITLE
Expected behaviour when rendering recipe for win64 using c_compiler

### ISSUE THREAD

**Issue by nichmor** [prefix-dev/rattler-build#733]
While working on `rattler-build` integration in `conda-smithy`, I've faced a following issue:

Having following `recipe.yaml`:

```
package:
    name: py-test
    version: 1.0.0
requirements:
    build:
      - if: win
        then:
          - ${{ compiler('c') }}
    host:
        - python
    run:
        - python
```

with these configurations:

```
c_compiler:     # [win]
        - vs2017        # [win]
        - vs2018        # [win]
  vc:             # [win]
        - '9'           # [win]
        - '14'          # [win]
 zip_keys:       # [win]
        - c_compiler    # [win]
        - python        # [win]
        - vc            # [win]
```

and trying to render the recipe for win64 on osx-arm64 I'm facing this error:

```
Target platform: win-64

 ╭─ Finding outputs from recipe
 │ Found 2 variants
 │ Build variant: py-test-1.0.0-py27h9778118_0
 │ 
 │ ╭─────────────────┬─────────╮
 │ │ Variant         ┆ Version │
 │ ╞═════════════════╪═════════╡
 │ │ c_compiler      ┆ vs2017  │
 │ │ python          ┆ 2.7     │
 │ │ target_platform ┆ win-64  │
 │ ╰─────────────────┴─────────╯
 │ Build variant: py-test-1.0.0-py35h2692361_0
 │ 
 │ ╭─────────────────┬─────────╮
 │ │ Variant         ┆ Version │
 │ ╞═════════════════╪═════════╡
 │ │ c_compiler      ┆ vs2018  │
 │ │ python          ┆ 3.5     │
 │ │ target_platform ┆ win-64  │
 │ ╰─────────────────┴─────────╯
 │
 ╰─────────────────── (took 0 seconds)

 ╭─ Resolving environments
 │ 
 │ Resolving environment for:
 │   Platform: osx-arm64
 │   Channels: 
 │    - /private/var/folders/rf/kyk8v0zd3993kvjg2p373plm0000gn/T/pytest-of-graf/pytest-33/test_py_matrix_on_azure_rattle0/output
 │    - conda-forge
 │   Specs:
 │    - vs2017_win-64
 │
 ╰─────────────────── (took 0 seconds)
Error:   × Failed to resolve dependencies: Cannot solve the request because of: No
  │ candidates were found for vs2017_win-64 *.
  │ 
  ╰─▶ Cannot solve the request because of: No candidates were found for
      vs2017_win-64 *.
      
```

For a conda-smithy user it will result in an error on rerendering process of the feedstock unless conda-smithy ( which run rattler-build build --render-only under the hood ) will be ran on win machine.

Is it an expected behaviour? Can we make resolving of requirements to "take" `target_platform` as `build_platform` in case of `--render-only`?

**Comment by wolfv** [comment]
This is fixed, right?

**Comment by nichmor** [comment]
yes, it is fixed by #744 ( using --target-platform )
